.pre-commit-config.yaml

```diff
@@ -141,3 +141,8 @@ repos:
           '--disable',
           'dangling-hyphen,line-too-long',
         ]
+  - repo: https://github.com/dnephin/pre-commit-golang
+    rev: v0.5.1
+    hooks:
+      - id: go-fmt
+      - id: go-imports
```

Comment:
arrow-adbc uses [golangci-lint](https://github.com/apache/arrow-adbc/blob/main/.pre-commit-config.yaml#L61), should we just use that instead?

Reply:
We could add that as well, but then I would need to fix all the issues it complains about.

Reply:
Fair, we can do that as a follow-up